Detectors/TRD/workflow/include/TRDWorkflow/GainCalibSpec.h

```diff
@@ -0,0 +1,168 @@
+    mCalibrator->process(dataGainCalib);
+    if (pc.transitionState() == TransitionHandlingState::Requested) {
+      LOG(info) << "Run stop requested, finalizing";
+      mRunStopRequested = true;
+      mCalibrator->checkSlotsToFinalize(o2::calibration::INFINITE_TF);
+      mCalibrator->closeOutputFile();
+    }
+    sendOutput(pc.outputs());
+  }
+
+  void endOfStream(o2::framework::EndOfStreamContext& ec) final
+  {
+    if (mRunStopRequested) {
+      return;
+    }
+    mCalibrator->checkSlotsToFinalize(o2::calibration::INFINITE_TF);
+    mCalibrator->closeOutputFile();
+    sendOutput(ec.outputs());
+  }
+
+  void stop() final
+  {
+    mCalibrator->closeOutputFile();
+  }
+
+ private:
+  std::unique_ptr<o2::trd::CalibratorGain> mCalibrator;
+  std::shared_ptr<o2::base::GRPGeomRequest> mCCDBRequest;
+  bool mRunStopRequested = false; // flag that run was stopped (and the last output is sent)
+  //________________________________________________________________
+  void sendOutput(DataAllocator& output)
+  {
+    // extract CCDB infos and calibration objects, convert it to TMemFile and send them to the output
+    // TODO in principle, this routine is generic, can be moved to Utils.h
+
+    using clbUtils = o2::calibration::Utils;
+    const auto& payloadVec = mCalibrator->getCcdbObjectVector();
+    auto& infoVec = mCalibrator->getCcdbObjectInfoVector(); // use non-const version as we update it
+    assert(payloadVec.size() == infoVec.size());
+
+    for (uint32_t i = 0; i < payloadVec.size(); i++) {
+      auto& w = infoVec[i];
+      auto image = o2::ccdb::CcdbApi::createObjectImage(&payloadVec[i], &w);
+      LOG(info) << "Sending object " << w.getPath() << "/" << w.getFileName() << " of size " << image->size()
+                << " bytes, valid for " << w.getStartValidityTimestamp() << " : " << w.getEndValidityTimestamp();
+
+      output.snapshot(Output{clbUtils::gDataOriginCDBPayload, "MEANDEDX", i}, *image.get()); // vector<char>
+      output.snapshot(Output{clbUtils::gDataOriginCDBWrapper, "MEANDEDX", i}, w);            // root-serialized
+    }
+    if (payloadVec.size()) {
+      mCalibrator->initOutput(); // reset the outputs once they are already sent
+    }
+  }
+};
+
+} // namespace calibration
+
+namespace framework
+{
+
+DataProcessorSpec getTRDGainCalibSpec()
+{
+  using device = o2::calibration::GainCalibDevice;
+  using clbUtils = o2::calibration::Utils;
+
+  std::vector<OutputSpec> outputs;
+  outputs.emplace_back(ConcreteDataTypeMatcher{o2::calibration::Utils::gDataOriginCDBPayload, "MEANDEDX"}, Lifetime::Sporadic);
+  outputs.emplace_back(ConcreteDataTypeMatcher{o2::calibration::Utils::gDataOriginCDBWrapper, "MEANDEDX"}, Lifetime::Sporadic);
+  std::vector<InputSpec> inputs;
+  inputs.emplace_back("input", "TRD", "GAINCALIBHISTS");
+  // inputs.emplace_back("calgain", "TRD", "CALIBGAIN", 0, Lifetime::Condition, ccdbParamSpec("TRD/Calib/CalibGain"));
```

Comment:
Here, the line is supposed to be used for retrieving the latest results from the ccdb (used by the function retrievePrev in CalibratorGain.cxx). However, since the ccdb entry is not yet created, it returns an error for now. So I assume it should be left commented for now, and get uncommented later. (the same goes for the function retrievePrev)